### DISCUSSION TITLE
Access to menus on official 7in touchscreen

### DISCUSSION THREAD

**Discussion by ceejayemm** [rern/rAudio#426] [Fixed Issues]
RPi 3B Plus with official PI 7in touch screen, RAudio-1 20210703, music library on NAS/NFS (OVM v5.x)

Updated to this version today to see if it would clear the issues noted in the title which seems to have been a problem for a while but I can't be sure when it started happening. As noted above I use this system via the built in web browser on the offical touchscreen.

The System/Network/Player/Features menu bar has disappeared, I can still access these options via the Cog Wheel menus (when using a remote browser from my PC - see below) but the menu bar does not appear at all.

When using the Touch Screen, on the Playback screen I generally use the CoverArt menu options but at the moment only the Stop and Play options are available to me via the touchscreen. In addition the Cog Wheel options are also unavailable to me at all. I can access the Library and Playlist screens via the bottom bar.

As I noted above I am not sure when this started but it is still happening today with the latest upgrade.

Thanks

Chris

**Comment by ceejayemm** [comment]
@rern 

Any thoughts on this problem please ?

Thanks

Chris

**Reply by rern** [reply]
Fixed.

**Reply by ceejayemm** [reply]
Rern

Thanks for the fixes so far, they make life a lot easier. I can now click on the options behind the Coverart and get to the setup functions. The bottom bar is back on the setup options too. Thanks you.

One small issue still remains. When I access the setup function pages I cannot get back to the Playback screen (or any of the main screens) because the blue 'X' in the top right hand corner of the touchscreen is not operative to touch. I have to use a PC based browser to access the installation and exit the setup functions from there.

Can this be fixed too please ?

Thanks

Chris

**Reply by ceejayemm** [reply]
Rern

Just seen that on the PC based browser when in the setup functions there is a Cog icon in the top right corner (there is a question mark icon instead) which seems to turn the sub function bottom bar on and off.  This cog icon does not appear on the RPi browser when on the touchscreen.

Chris

**Reply by ceejayemm** [reply]
Situation seems a bit better after applying 200210724, although I didn't notice anything to this effect in the release notes, in that I can now get to to the Setup screens and other controls by touching the central Coverart display and then touching the Cog wheel displayed there. The Cog wheel in the top right hand corner is still unusable.

I can now see the submenu bar across the bottom of the screen (for System, Networks, Player, Features) however I can only return to the main screen from Networks as it is the only page which doesn't have a scrollbar on the right hand side. It seems the scrollbar functionality overrides the access to the blue 'X' on the System, Player and Features subpages which prevents its use and thus returning to the main screens. The Help question mark on these same pages also suffers from the same problem.

Just to be clear this is only noticeable when using rAudio-1 on the official RPi 7in touchscreen, all screens work as I would expect when accessing these pages from a normal browser (for me, MS Edge) on a Windows 10 PC.

Regards.

Chris

**Reply by ceejayemm** [reply]
Rern

Applied 20210731 today and this seems to have reverted some of the gains made with the earlier versions. With this version on the PI Official Touchscreen I can no longer access any of the Setup options. I can access the Playback, Library and Playlist main screens but all touch access via the Coverart on the Playback screen has completely disappeared. The Cog in the top right corner also now does not work.

Chris

**Reply by ceejayemm** [reply]
Rern

I tried resetting the Browser on RPI settings, the Screen Off setting now sets itself to 36000mins (where I origianlly had it set to 600). For the other issues, might this be a clue ?

# pacman -Q chromium; systemctl status localbrowser

chromium 89.0.4389.90-1
● localbrowser.service - Local Browser - Chromium
     Loaded: loaded (/etc/systemd/system/localbrowser.service; enabled; vendor preset: disabled)
     Active: active (running) since Sat 2021-07-31 17:07:18 UTC; 2s ago
   Main PID: 121837 (sudo)
      Tasks: 44 (limit: 1056)
     CGroup: /system.slice/localbrowser.service
             ├─121837 /usr/bin/sudo /usr/bin/startx
             ├─121840 /bin/sh /usr/bin/startx
             ├─121856 xinit /etc/X11/xinit/xinitrc -- /etc/X11/xinit/xserverrc :0 -auth /tmp/serverauth.XzzDIP5uLb
             ├─121857 /usr/lib/Xorg -nolisten tcp :0 -auth /tmp/serverauth.XzzDIP5uLb
             ├─121887 /bin/bash /etc/X11/xinit/xinitrc
             ├─121892 matchbox-window-manager -use_cursor no
             ├─121893 /usr/lib/chromium/chromium --app=http://localhost --kiosk --incognito --no-sandbox --no-xshm --force-device-scale-factor=0.7
             ├─121955 /usr/lib/chromium/chromium --type=zygote --no-zygote-sandbox --no-sandbox --force-device-scale-factor=0.7
             └─121956 /usr/lib/chromium/chromium --type=zygote --no-sandbox --force-device-scale-factor=0.7

Jul 31 17:07:18 RAConservatory sudo[121857]:         Before reporting problems, check http://wiki.x.org
Jul 31 17:07:18 RAConservatory sudo[121857]:         to make sure that you have the latest version.
Jul 31 17:07:18 RAConservatory sudo[121857]: Markers: (--) probed, (**) from config file, (==) default setting,
Jul 31 17:07:18 RAConservatory sudo[121857]:         (++) from command line, (!!) notice, (II) informational,
Jul 31 17:07:18 RAConservatory sudo[121857]:         (WW) warning, (EE) error, (NI) not implemented, (??) unknown.
Jul 31 17:07:18 RAConservatory sudo[121857]: (==) Log file: "/var/log/Xorg.0.log", Time: Sat Jul 31 17:07:18 2021
Jul 31 17:07:18 RAConservatory sudo[121857]: (==) Using config directory: "/etc/X11/xorg.conf.d"
Jul 31 17:07:18 RAConservatory sudo[121857]: (==) Using system config directory "/usr/share/X11/xorg.conf.d"
Jul 31 17:07:18 RAConservatory sudo[121873]: modprobe: FATAL: Module g2d_23 not found in directory /lib/modules/5.10.25-4-ARCH

Chris

**Comment by ceejayemm** [comment]
Rern

This is still an issue, even after updating to 20210807. My wife likes to use the touchscreen, it is normally easier for her to do so than to connect up with an external browser (where all works as it should). I can touch the screen when nothing is displayed to 'wake up' the display and then I can use the main function menu items (Library, Playback and Playlist), stop and play but NOTHING else works via the touch screen

I hope this can be sorted out soon.

Thanks

Chris

**Comment by ceejayemm** [comment]
Rern

I have now re-imaged the SD card (using i20210716) and then updated to 20210813 and still the problems with the touchscreen access persist. Is this likely to be fixed any time soon or do I just need to accept the limited touchscreen functionality is what it is now ? This latter point would be a real shame if this is the future as it seems a backward step.

BTW - I see that you have just released i20210813, will reflashing the SD card to this version be any use given that this is effectively where I am already  with updates ?

Chris

**Reply by rern** [reply]
I could not reproduce your issue on my 3.5" TFT LCD.
Any screenshots?

**Reply by ceejayemm** [reply]
Rern

For info: RPi 3B+ with 7in official touchscreen, running i20210716, updated to 20210813. All screens work correctly when accessed via remote browser (eg Chrome/ MS Edge) from Win 10 PC.

On the local touch screen all main screens (ie Library, Playback and Playlist) display correctly. 

On Playback screen, when playing a web radio, only Stop, Play and Volume buttons respond to touch screen action. When playing albums then stop, play, forward, rewind and pause are shown and are operational. All other touch actions (eg Coverart options, circular Volume control, Options Cog Wheel, Help Question Mark etc) are totally unresponsive.  Swipe Left or Right is not available. Sometimes - and I am not sure how this happens, maybe after multiple touches - pressing on the Coverart logo produces a tiny RAudio logo in the bottom right hand corner of the Coverart logo.

On the Playback options I have Top-Bottom Bars, Bars Always On, Coverart, Volume, Buttons and Progress Bar are all turned ON, all other options are turned OFF. The 'Default if No Coverart is Found' is set to rAudio.

On the Library screen all Library buttons are displayed and work to some extent but this can be hit and miss, for example some times I can choose a new web radio station but I can't add or replace it to the Playlist then other times I can do this. Cog Wheel and Help options not available.

Playlist screen - similar to Library screen. Sometimes options work other times they don't. Cog Wheel and Help not available.

On the few times I have been able to get to the Setup options, not recently though, the new System / Networks/ Player/ Features menu bars are not displayed on the bottom of the screen.

I have attached some, not very good, screenshots. They only show the standard screens as you would expect and cannot show the problems I am having.

I hope this helps.

Chris
[Camera Roll.zip](https://github.com/rern/rAudio-1/files/6986718/Camera.Roll.zip)

**Comment by ceejayemm** [comment]
Rern

Update to 20210820 doesn't improve the situation, but then there was nothing in the release notes to suggest it would. I don't know what else I can tell you about this issue as no obvious errors are generated and the screen layouts etc appear to be correct BUT they just don't response to touch as they should. If you need me to do any local diagnostics or testing doing then please let me know. The touch screen itself seems responsive as a touch anywhere will awake the screen from 'sleep' mode.

Chris

**Reply by rern** [reply]
Your touch screen might need a calibration. 
Unfortunately, there's a GUI for 3.5" LCD only but `xinput_calibrator` command should be the same and already installed.

**Comment by ceejayemm** [comment]
Rern

How do I run this command as when I try I get this message:

RAConservatory:~  #  xinput_calibrator
Unable to connect to X server

 I presume I need to start and X Server but how ?

Thanks

Chris

**Comment by ceejayemm** [comment]
Rern

Still no luck with xinput_calibrator

To 'try to do something' to help my wife out who is the main user of this interface, I flashed a new USB drive with the latest 64bit image (i20210813) and then updated to 20210820. This helped a little in that I can now access the Coverart options menu by double clicking on the Coverart image. Once there I can used the menus, sometimes via a single click at others via a double click. I can't however exit from these menu back to the main pages.

So I think the thoughts about needing to calibrate the touchscreen might be a red herring. Is there anything else I can try as I am loath to ditch this touchscreen which until recently has worked flawlessly with Rune/ rAudio versions.

Chris

**Comment by R101G** [comment]
@ceejayemm Are you using a wired or a wifi connection? As nobody else has reported the problem, either you are the only person using the 7" screen, or there is something different about your setup.

I am putting some of the mysterious problems I have had down to incomplete updates caused by the wifi clock sync problem. If you are using wifi, check the time and date are correct.

**Reply by ceejayemm** [reply]
@rern / @R101-Github 

Yes, I am using a RPi 3B+ and wifi. Yes The system time stamp was wrong on my new installation, in my case Aug 13th but can't remember actual time. I ran systemctl status systemd-timesyncd as detailed in https://github.com/rern/rAudio-1/discussions/472 which seemed to correct the date/time. I ran pacman -Syu to update the installation which seemed to go OK and then I created the file /etc/systemd/network/wireless.network as also noted in that post and rebooted.  At this point the RPi became completely inaccessible and I am now at the stage of having to re-flash the card again. This is the reason why I am a bit vague about some of the details above having nothing available to refer to. 

Chris

**Comment by R101G** [comment]
@ceejayemm The status command you ran was requested by rern, but I do not understand why it would affect the time setting. Entering the restart systemd-networkd command a couple of times got the time right for me until the next reboot (I don't know how, but others report the same thing), and adding the additional file has made it survive rebooting. I used a wired connection to install, which meant the time was OK until I went back to wifi.

I do not know what went wrong with your reboot. I don't see how the additional file would cause any more problem than the existing ones, but not much about this makes sense to me. I can only suggest getting the time to stick reliably before doing any any updates.

**Comment by ceejayemm** [comment]
@rern / @R101-Github 

Strange. On reboot the PI will not connect to the wifi with this revised setup. If I put the 'wifi' file back on the card and reboot then wifi is available until the next reboot.  If I don't replace the 'wifi' file I have to go into Networks reconnect to the existing Network (which is remembered but obviously disconnected).

With wifi connected ,  systemctl status systemd-timesyncd shows:

RAConservatory:~  #  systemctl status systemd-timesyncd
* systemd-timesyncd.service - Network Time Synchronization
     Loaded: loaded (/usr/lib/systemd/system/systemd-timesyncd.service; enabled; vendor preset: enabled)
     Active: inactive (dead)
       Docs: man:systemd-timesyncd.service(8)

Starting systemd-timesyncd followed by systemctl status systemd-timesyncd shows:

RAConservatory:~  #  systemctl start systemd-timesyncd
RAConservatory:~  #  systemctl status systemd-timesyncd
* systemd-timesyncd.service - Network Time Synchronization
     Loaded: loaded (/usr/lib/systemd/system/systemd-timesyncd.service; enabled; vendor preset: enabled)
     Active: active (running) since Thu 2021-08-26 14:34:44 UTC; 3min 50s ago
       Docs: man:systemd-timesyncd.service(8)
   Main PID: 3009 (systemd-timesyn)
     Status: "Initial synchronization to time server 81.21.65.169:123 (pool.ntp.org)."
      Tasks: 2 (limit: 1056)
        CPU: 220ms
     CGroup: /system.slice/systemd-timesyncd.service
             `-3009 /usr/lib/systemd/systemd-timesyncd

Aug 26 14:34:44 RAConservatory systemd[1]: Starting Network Time Synchronization...
Aug 26 14:34:44 RAConservatory systemd[1]: Started Network Time Synchronization.
Aug 26 14:37:14 RAConservatory systemd-timesyncd[3009]: Initial synchronization to time server 81.21.65.169:123 (pool.ntp.org).
RAConservatory:~  #

systemctl enable systemd-timesyncd followed by a further reboot results in disconnected wifi network on startup.

Removing /etc/systemd/network/wireless.network and rebooting results in system starting normally, with network remembered and started and:

RAConservatory:~  #  systemctl status systemd-timesyncd
* systemd-timesyncd.service - Network Time Synchronization
     Loaded: loaded (/usr/lib/systemd/system/systemd-timesyncd.service; enabled; vendor preset: enabled)
     Active: active (running) since Thu 2021-08-26 14:41:28 UTC; 6min ago
       Docs: man:systemd-timesyncd.service(8)
   Main PID: 275 (systemd-timesyn)
     Status: "Daemon is running"
      Tasks: 1 (limit: 1056)
        CPU: 201ms
     CGroup: /system.slice/systemd-timesyncd.service
             `-275 /usr/lib/systemd/systemd-timesyncd

Aug 08 18:55:47 RAConservatory systemd[1]: Starting Network Time Synchronization...
Aug 08 18:55:47 RAConservatory systemd-timesyncd[275]: System clock time unset or jumped backwards, restoring from recorded timesta>
Aug 26 14:41:28 RAConservatory systemd[1]: Started Network Time Synchronization.
lines 1-14/14 (END)
^C
RAConservatory:~  #

So maybe there is a start up difference between RPi3B+ and RPi4B which needs to be looked at ?

With known good time setting I ran pacman -Syu again which updated 19 components. System now back to original problem where only way to access setup options is to double click the Coverart logo and then double tap the Cog icon from there.   Once into setup I can only return to main menu from Networks (where there is no scroll bar present). Trying to return to main screens from System, Player and Feature (where scrollbars are present is still not possible, This is  despite having 'Bars always on' turned off on the Playback context menu.

Chris

**Comment by R101G** [comment]
@ceejayemm I tried my 64bit SD card in a pi3, and it worked, so that does not seem to be the problem. I think @teslasparks301 has confirmed that adding the file does work. Are you sure you do not have a format error in your wireless.network file? 

Not that updating with the clock sync'd seems to have helped with your touchscreen problem :)

**Comment by ceejayemm** [comment]
Rern

I am afraid update 20210827 does not help this problem.

Chris

**Comment by ceejayemm** [comment]
@rern 

The information on #491 (by @klassikmann) struck a chord (no pun intended) with me. I used the Browser on RPi settings to rotate the display by 90 degrees (doesn't matter whether this is clockwise or anti-clockwise) and hey presto the problems I have been having with the touchscreen disappeared. The only problem with this 'fix' is the crick in the neck I will get both from looking at the screen in this manner and from the wife who won't like it like this :-)

If this now helps to get this fixed that would be great. The only issue outstanding then will be the Browser on RPi timeout setting which is getting values of between 60000 and 466560000000 secs on screen when I have been using 600 sec. Despite the errant values the timeout does actually seem to be set to 600 sec somewhere behind the scenes.

Chris

**Comment by ceejayemm** [comment]
Rern

No improvement with 202109032. It seems to be related to proximity of the Cog icon in the top right corner to the right hand scroll bars as shown by:

a) problems occur on System, Player and Features pages (which all have scrollbars) but NOT on Networks (which doesn't)
b) rotating screen (via Browser on RPI, 90 deg (clockwise or anticlockwise) solves the problems - but give me a bad neck 

in the case of ((b) the scroll bars are much reduced if not removed in this orientation.

If I use the Coverart underlying icon layer, I still need to double tap to get these to work and also (on Browser on RPi setup option) the timeout display setting varies wildly from 36000 to 466560000000 as previously noted.

Chris

**Comment by ceejayemm** [comment]
Rern

Update 20210911 has no effect of this issue.  I have split off the 'Browser on RPi' settings problem to a new ticket (#510).

Chris

**Comment by ceejayemm** [comment]
RPi3B Plus, RAudio-1 i20210911 64bit installed, Wifi connection (DHCP, fixed IP via router)
Applied 20210924 and rebooted.

Browser on RPi Screen Off and Zoom settings now seem to work as they did in the past.

This update seems to have no effect on the other problems I have previously reported using RAudio-1 with the Official RPi 7in Touch screen.

Chris

**Comment by ceejayemm** [comment]
@rern 

I saw that in #536 you asked @sagnasty who seems to be having a similar problem to me to give you the output of:

player-data.sh | jq .

I have done this for this problem too, see:

RAConservatory:~  #  player-data.sh | jq .
{
  "page": "player",
  "devices": [
    {
      "aplayname": "merus-amp",
      "card": 0,
      "device": 0,
      "dop": 0,
      "hw": "hw:0,0",
      "hwmixer": "A.Mstr Vol",
      "mixers": 4,
      "mixerdevices": [
        "A.Mstr Vol",
        "B.L Vol",
        "C.R Vol",
        "D.Lim thresh"
      ],
      "mixermanual": false,
      "mixertype": "hardware",
      "name": "MERUS Audio Amp"
    }
  ],
  "active": true,
  "asoundcard": 0,
  "audioaplayname": "merus-amp",
  "audiooutput": "MERUS Audio Amp",
  "autoupdate": false,
  "buffer": false,
  "bufferconf": 4096,
  "bufferoutput": false,
  "bufferoutputconf": 8192,
  "counts": {
    "album": 318,
    "albumartist": 126,
    "artist": 563,
    "composer": 1011,
    "conductor": 2,
    "date": 153,
    "genre": 34,
    "nas": 1,
    "sd": 0,
    "usb": 0,
    "song": 4815,
    "webradio": 25
  },
  "crossfade": false,
  "crossfadeconf": 1,
  "custom": false,
  "equalizer": false,
  "ffmpeg": true,
  "normalization": false,
  "replaygain": false,
  "replaygainconf": "auto",
  "soxr": false,
  "soxrconf": [
    20,
    50,
    91.3,
    100,
    0,
    0
  ],
  "version": "0.22.11-1"
}
RAConservatory:~  #

I hope this helps to move this problem forward too.

Chris

**Comment by ceejayemm** [comment]
RPi 3B+, 32 bit i20210911 base, with Raspberry Official 7in touchscreen. All of the comments below relate to access via the local 7in touchscreen. Access from a Windows 10 PC via MS Edge or Chrome (latest versions) all work correctly on all pages.

Some progress after applying today's 20211010 update. 

Coverart menus on Playback screen now work with a single click

Cog menu (top right on Playback page) when clicked has no effect, but Cog menu items can be accessed via Cog in Coverart menu.

System, Player and Feature pages display as expected BUT have a vertical scroll bar on the right hand side.  It is not possible to exit via 'X' from these screen to any other page or access '?' Help from any of these pages. Touch swipe/scrolling does work so the right hand scrollbars seem redundant.

The Network page DOES NOT have the right hand scrollbar, on this page the 'X' exit and '?' help icons work correctly.

It seems that the Cog icon, the 'X' exit and '?' help on the affected pages all suffer from problems when the right hand scrollbar (despite scrollbars being turned off in setup) is being displayed.  Would it be possible to create a temporary version for testing where the Cog, 'X' exit and '?' help icons are moved to the left hand side (or further left than the current positions) and/or the scrollbars actually turned off according to the settings to see if this corrects these problems ? 

This has been going on for a long time and it would be good to get this sorted once and for all.  I am happy to try test versions if that would help.

Thanks

Chris

**Reply by rern** [reply]
Try:
- Features > Browser on RPi > enable mouse pointer
- Use a stylus to move the pointer around to verify the tip and the pointer are at the same position.

**Comment by ceejayemm** [comment]
@rern 

Updated to 20211011 and enabled mouse pointer as suggested. Using a stylus I can see that the pointer correctly locates the icons in the Coverart menu (for example) - see Pic1.jpg. Clicking the Cog icon on the Coverart menu opens the Settings menu and I can click on a submenu item to activate the submenu.  

If I click directly on the Cog icon on the Playback screen the mouse pointer is to the right of the Cog (see pic4.jpg) and nothing happens.

On the System (and Player/Features) menu clicking the 'X' exit icon also places the pointer to the right of the 'X' icon on the visible scrollbar (pic5.jpg). Clicking on the '?' help icon seems to leave the mouse point where it last was (ie no discernible movement) (pic3.jpg).

All pictures in attached zip file, I hope this helps. 

Thanks for looking into this further. Please let me know if you need me to do anything else.

Chris

[pics.zip](https://github.com/rern/rAudio-1/files/7323411/pics.zip)

**Comment by ceejayemm** [comment]
@rern 

No obvious improvement after flashing a new card with the 64bit i20211015. Further to the pictures I previously sent, I have noticed that the mouse pointer cannot be moved into a column which stretches from the top of the screen all the way down to the bottom. This column is about 0.5cm in from the right hand edge of the screen and about 1cm in width. 

Chris

**Comment by ceejayemm** [comment]
Changed title to better reflect current state of this issue.

Applied update 20211019 which appears to switch local browser from Chromium to Firefox as per release notes. Problem still exists with column on right hand side of screen where the mouse pointer cannot be moved into on the local screen thus not able to access menus from Cog icon.

Chris

**Reply by rern** [reply]
Try: [Raspberry pi 7 inch touch screen calibration](https://www.thingsconnected.io/raspberry-pi-7-inch-touch-screen-calibration/)

**Comment by ceejayemm** [comment]
Updated to i20211022, still same problem.  Tried all sorts to get xinput_calibrator working to no avail.  After more searching I came across ts_calibrator / ts_test which seem to be already installed in this version of Arch Linux.  Running ts_calibrator gives me a test screen and subsequently produces a file '/etc/pointercal' with the results of the testing I did, so some progress.  However when I reboot the PI to apply the new settings the PI / rAudio-1 would not complete the boot process.  Once I removed '/etc/pointercal' the PI booted normally.

Unless anybody has any better ideas I think I am going to have to give up on this touchscreen and give my wife a tablet to access the app via the normal web url. 

Chris

**Comment by ABH10** [comment]
I installed i20220317 on an RPi3B+ with official 7" touchscreen and I experience exactly the same thing. The menu lines in the top and bottom are not visible. Funny thing is, that the bottom menu seems to be working although not visible. The top menu is not working and is not visible. The WEB UI has the top and bottom menu lines.
![image](https://user-images.githubusercontent.com/22498335/160228353-bf40ffc2-5745-4d71-8321-ccd25f2a16cd.png)

**Reply by ceejayemm** [reply]
I don't have a fix for this but:

- have you turned 'On' the Playback Options  'Bars Always on' 
- set the Features / Browser on RPi / Browser Display option to 70% (or thereabouts)

Might help.

Out of interest, can you close option screens which have a scrollbar (System, Player, Features) on your touchscreen as I still have problems with this.

Chris

Chris

**Comment by rern** [comment]
Tap top of coverart for guide, the funny thing:
![guide](https://user-images.githubusercontent.com/10094236/160267930-a46eac4f-f0ef-4bbc-844e-47e45ea149ab.jpg)
